types/ssh_config.go

```diff
@@ -0,0 +1,23 @@
+package types
+
+type PubkeyAuthValue string
+
+const (
+	PubkeyAuthValueYes       PubkeyAuthValue = "yes"
+	PubkeyAuthValueNo        PubkeyAuthValue = "no"
+	PubkeyAuthValueHostBound PubkeyAuthValue = "host-bound"
+	PubkeyAuthValueUnbound   PubkeyAuthValue = "unbound"
+)
+
+func (p PubkeyAuthValue) String() string {
+	return string(p)
+}
+
+// SSHConfig is the SSH client configuration that a clab node requires.
+type SSHConfig struct {
+	PubkeyAuthentication PubkeyAuthValue
+}
+
+func NewSSHConfig() *SSHConfig {
+	return &SSHConfig{}
+}
```

Comment:
@steiler I went with removing one extra layer here -- KindSpecifics
I think we are fine without it right now, because SSHConfig might be different for the same kind based on the OS version.
So it felt wrong using this struct name. I just stripped it, and kept the rest.

Reply:
Yeah I wanted to discuss pulling the license policy and host requirements under the kind specifics, which to me seemed to be kinda logical. But yes, this works as well.